### PR TITLE
[FIX] mrp_workorder: prevent null value for date_to

### PR DESCRIPTION
**Pull request by myhy-odoo**
Problem: When the user inputs a value greater than 490,000 minutes for expected duration on a workorder, a Validation Error is thrown signifying that date_to is null but it's a required field when "Produce All" button is clicked. The workorder's date_finished gets recalculated always so it'll evaluate to False with a large value.

Purpose: To stay consistent with v16, the ValidationError should not occur and date_finished should not be recalculated if a value exists already. Produce All should execute smoothly

fix based on v16: https://github.com/odoo/odoo/blob/4d154a9f08cb2204df8566ed5c1e4f5e59e5d212/addons/mrp/models/mrp_workorder.py#L273

Steps to Reproduce on Runbot:
1. Install MRP
2. Enabled Work Orders in Settings > Manufacturing
3. Input a value greater than 490,000 on a workorder for expected duration
4. Click "Produce All" and ValidationError is thrown

opw-3959230
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by myhy-odoo** [comment]
Hello @ticodoo! Can you review this PR? Thanks in advance!

**Comment by ethanrobv** [comment]
It might be better to enforce the max duration rather still letting the user write some arbitrarily large value which will then not get reflected in the date_finished field (or at the very least notify them of the discrepancy)

**Comment by ticodoo** [comment]
Hello @myhy-odoo 

Unfortunately this fix only handles 1 use case flow and seems like a bit of a bandaid that hides the underlying problem. Also, I think your fix will break the end date calculation/recalculation and/or the (real) `duration`

It seems like there's an issue with the `_calculate_date_finished` where `self.workcenter_id.resource_calendar_id.plan_hours(...` not finding suitable intervals. Note that this calculation depends on the working hours (`resource_calendar_id`) of the work center as well, so the maximum expected duration without this flow breaking can vary.

Note that a similar issue happens with a large expected duration when (note that the plan button fails at a smaller value):
- the MO "Plan" button is clicked
- A Start Date is set, but not an End Date + try to save
- when you try to start the WO timer in the shop view (this one throws a very confusing error about unplanning)
- click on the "Plan" button in the WO's line (same error as the shop floor)

but the issue in the "Plan" use case is due to `_get_first_available_slot`  in `mrp_workcenter` having a similar issue calculating reasonable time slots.

I would say in all cases it's maybe better if you could either:
- make it so that larger numbers can always be handled (this might not be possible, not sure since `resource` is handled by another team and I've never used `partial()`  before so I don't know much about it)
- return a usererror about the expected duration being too large for the current working hours of the workcenter it's assigned to

TLDR: this is a pretty complicated bug. If it's too much to investigate feel free to pass it along to bugfix

cc: @ethanrobv

**Comment by ethanrobv** [comment]
I am currently working on an interval calculation rework here: https://github.com/odoo/odoo/pull/180252
which will fix this issue if I can eventually get it passing remaining tests and approved.
